packages/@eventual/core-runtime/src/local/stores/bucket-store.ts

```diff
@@ -47,14 +50,14 @@ export class LocalBucketStore implements BucketStore, LocalSerializable {
           const { body, ...meta } = value;
           return [
             [
-              key,
+              `${key}${OBJECT_SUFFIX}`,
```

Comment:
I don't understand why this is useful? Does it not cause more problems?

Reply:
There would be an issue if the user named their item `[key]/##object` while also having a `[key]`. It seems like a fair limitation for now.

Right now we fail if a user creates anything nested `[key]` and `[key]/[subkey]`.

Reply:
We could borrow the hash technique from the cdk?

Reply:
I'm fine with this limitation also though. Would prefer easy to read paths locally containing all of the files over 100% correctness

Reply:
Is there a way we can avoid always adding it? It should be trivial to browse the buckets using the local file system.

Reply:
Well we always have the meta anyways, so I kind of like the 

[key]/##object
[key]/##meta

instead of 

[key]
[key]##meta

I had before

Though I agree it would be nice to be to able to use file extensions in keys.

To handle it dynamically, we'd need to check if there are any overlapping keys and switch the strategy based, then support both on load.

Reply:
Updated
![image](https://github.com/functionless/eventual/assets/289213/5a9a9542-fb25-4a18-8701-572e884d2163)

Reply:
with extension - [bucket]/[path]/[last path segment]-[hash].[ext]
without extension - [bucket]/[path]/[last path segment]-[hash]
meta - [bucket]/[path]/[last path segment]-[hash].eventual.meta.json

Reply:
Goals:
1. maintain the file system structure
2. ensure that keys can also be path prefixes like s3 (added the hash to decrease chance of collisions)
3. preserve file extension if present so the file opens correctly in ides/OSs
4. co-locate the meta with the object
5. maintain the file name in the file
6. ensure that files do not change between runs (hash and not uuid)